src/cli/templates/_worker.js/index.ts

```diff
@@ -0,0 +1,58 @@
+import { getCloudflareGlobalContextAls } from '../../../utils/utils/cloudflareGlobalContext';
+import { handleRequest } from './handleRequest';
+import { adjustRequestForVercel } from './utils';
+import type { AsyncLocalStorage } from 'node:async_hooks';
+
+declare const __NODE_ENV__: string;
+
+declare const __CONFIG__: ProcessedVercelConfig;
+
+declare const __BUILD_OUTPUT__: VercelBuildOutput;
+
+declare const __ENV_ALS_PROMISE__: Promise<null | AsyncLocalStorage<unknown>>;
+
+export default {
+	async fetch(request, env, ctx) {
+		const envAsyncLocalStorage = await __ENV_ALS_PROMISE__;
+		if (!envAsyncLocalStorage) {
+			return new Response(
+				`Error: Could not access built-in Node.js modules. Please make sure that your Cloudflare Pages project has the 'nodejs_compat' compatibility flag set.`,
+				{ status: 503 }
+			);
+		}
+
+		const cloudflareGlobalContextAls = getCloudflareGlobalContextAls();
+
+		if (!cloudflareGlobalContextAls) {
+			return new Response(
+				`Internal Server Error: cannot retrieve the cloudflare global context AsyncLocalStorage`,
+				{ status: 500 }
+			);
+		}
+
+		return cloudflareGlobalContextAls.run(
+			{
+				cf: request.cf,
+				ctx,
+			},
+			() => {
+				return envAsyncLocalStorage.run(
```

Comment:
instead of this nesting `run` we could use `enterWith` but it seems like the latter is experimental: https://nodejs.org/api/async_context.html#asynclocalstorageenterwithstore (and I am not sure if we even support that) so I personally would not go with that for now (also the nesting is not too too bad in my opinion 🙂)

anyways I can experiment with `enterWith` if someone feels strongly about it (but possibly I'd do in a separate PR and keep this one just focused on the utils logic)

cc. @GregBrimble